src/features/class-full-page-cache-404.php

```diff
@@ -0,0 +1,337 @@
+<?php
+/**
+ * Class file for Full Page Cache for 404s.
+ *
+ * (c) Alley <info@alley.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @package Alley\WP\WP_404_Caching
+ */
+
+declare( strict_types=1 );
+
+namespace Alley\WP\WP_404_Caching\Features;
+
+use Alley\WP\Types\Feature;
+use WP_Query;
+use function defined;
+use function function_exists;
+
+/**
+ * Full Page Cache for 404s.
+ */
+final class Full_Page_Cache_404 implements Feature {
+
+	/**
+	 * Cache group.
+	 *
+	 * @var string
+	 */
+	public const CACHE_GROUP = 'wp_404_caching';
+
+	/**
+	 * Cache key.
+	 *
+	 * @var string
+	 */
+	public const CACHE_KEY = '404_cache';
+
+	/**
+	 * Cache key for stale cache.
+	 *
+	 * @var string
+	 */
+	public const STALE_CACHE_KEY = '404_cache_stale';
+
+	/**
+	 * Stale cache time.
+	 *
+	 * @var int
+	 */
+	public const STALE_CACHE_TIME = DAY_IN_SECONDS;
```

Comment:
This appears to be a duplicate of `DEFAULT_STALE_CACHE_TIME` that I don't see used anywhere

```suggestion
```

Reply:
Will fix!